File: movemork/Features/Welcome/WelcomeEvidenceStack.swift
//
//  WelcomeEvidenceStack.swift
//  movemork
//
//  Welcome-only hero — saved evidence on a subtle report sheet.
//

import SwiftUI

struct WelcomeEvidenceStack: View {
    var imageName: String = "welcome-kitchen-main"
    var photoHeight: CGFloat = 196
    var tagsVisible: Bool = true

    private let issueCallouts = ["Already there", "Chipped paint"]
    private let sheetOffset: CGFloat = 10
    private let cardRadius: CGFloat = 28
    private let photoRadius: CGFloat = 23

    @Environment(\.accessibilityReduceMotion) private var reduceMotion

    var body: some View {
        ZStack(alignment: .topLeading) {
            reportSheetLayer
            evidenceCard
        }
    }

    /// Neutral paper sheet — charcoal, not green; slight offset behind evidence.
    private var reportSheetLayer: some View {
        RoundedRectangle(cornerRadius: cardRadius - 2, style: .continuous)
            .fill(MoveMarkTheme.Colors.paperSurface.opacity(0.12))
            .overlay(
                RoundedRectangle(cornerRadius: cardRadius - 2, style: .continuous)
                    .stroke(MoveMarkTheme.Colors.cardStroke.opacity(0.35), lineWidth: 0.75)
            )
            .padding(.top, sheetOffset + 6)
            .padding(.leading, sheetOffset + 4)
            .padding(.trailing, 2)
            .padding(.bottom, 2)
    }

    private var evidenceCard: some View {
        VStack(spacing: 0) {
            photoPane
            receiptFooter
        }
        .background(MoveMarkTheme.Colors.evidenceCard)
        .overlay(
            RoundedRectangle(cornerRadius: cardRadius, style: .continuous)
                .stroke(MoveMarkTheme.Colors.cardStroke.opacity(0.9), lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: cardRadius, style: .continuous))
        .shadow(color: Color.black.opacity(0.22), radius: 10, x: 0, y: 6)
    }

    private var photoPane: some View {
        ZStack(alignment: .topLeading) {
            ZStack {
                Image(imageName)
                    .resizable()
                    .scaledToFill()
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .clipped()

                VStack {
                    Spacer(minLength: 0)
                    LinearGradient(
                        colors: [.clear, Color.black.opacity(0.34)],
                        startPoint: .top,
                        endPoint: .bottom
                    )
                    .frame(height: 48)
                }
            }
            .clipShape(RoundedRectangle(cornerRadius: photoRadius, style: .continuous))

            Text("Kitchen · Apr 14 · 5:42 PM")
                .font(.system(size: 10, weight: .medium, design: .monospaced))
                .foregroundStyle(Color.white.opacity(0.92))
                .lineLimit(1)
                .minimumScaleFactor(0.85)
                .padding(.horizontal, 8)
                .padding(.vertical, 5)
                .background(
                    Color.black.opacity(0.55),
                    in: RoundedRectangle(cornerRadius: 6, style: .continuous)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: 6, style: .continuous)
                        .stroke(Color.white.opacity(0.12), lineWidth: 0.5)
                )
                .padding(.leading, 12)
                .padding(.top, 12)

            if tagsVisible {
                VStack(alignment: .leading, spacing: 5) {
                    ForEach(Array(issueCallouts.enumerated()), id: \.offset) { index, label in
                        Text(label)
                            .font(.system(size: 9.5, weight: .medium))
                            .foregroundStyle(Color.white.opacity(0.9))
                            .lineLimit(1)
                            .padding(.horizontal, 7)
                            .padding(.vertical, 3)
                            .background(
                                Color.black.opacity(0.62),
                                in: RoundedRectangle(cornerRadius: 5, style: .continuous)
                            )
                            .overlay(
                                RoundedRectangle(cornerRadius: 5, style: .continuous)
                                    .stroke(Color.white.opacity(0.14), lineWidth: 0.5)
                            )
                            .scaleEffect(reduceMotion ? 1 : (tagsVisible ? 1 : 0.94))
                            .animation(
                                reduceMotion ? nil : .easeOut(duration: 0.32).delay(Double(index) * 0.05),
                                value: tagsVisible
                            )
                    }
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottomLeading)
                .padding(.leading, 12)
                .padding(.bottom, 12)
                .padding(.trailing, 8)
            }
        }
        .frame(maxWidth: .infinity)
        .frame(height: photoHeight)
        .padding(.horizontal, 12)
        .padding(.top, 12)
    }

    private var receiptFooter: some View {
        HStack(alignment: .top, spacing: 8) {
            Image(systemName: "checkmark.circle.fill")
                .font(.system(size: 18, weight: .semibold))
                .foregroundStyle(MoveMarkTheme.Colors.primary.opacity(0.85))
                .padding(.top, 1)

            VStack(alignment: .leading, spacing: 2) {
                Text("Saved proof record")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(MoveMarkTheme.Colors.textPrimary)
                    .lineLimit(1)

                Text("12 photos attached · Move-in inspection")
                    .font(.system(size: 11.5, weight: .medium))
                    .foregroundStyle(MoveMarkTheme.Colors.textMuted)
                    .lineLimit(2)
                    .fixedSize(horizontal: false, vertical: true)
            }

            Spacer(minLength: 4)

            Text("Report-ready")
                .font(.system(size: 10, weight: .medium))
                .foregroundStyle(MoveMarkTheme.Colors.textMuted.opacity(0.85))
                .lineLimit(1)
                .padding(.top, 2)
        }
        .padding(.horizontal, 14)
        .padding(.vertical, 11)
        .background(MoveMarkTheme.Colors.fieldFill.opacity(0.38))
    }
}

File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/welcome/WelcomeEvidenceStack.kt
package com.surensureshkumar.movemark.features.welcome

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.surensureshkumar.movemark.R
import com.surensureshkumar.movemark.core.design.MMColors
import com.surensureshkumar.movemark.core.design.MMMotion

private val issueCallouts = listOf("Already there", "Chipped paint")

/**
 * Welcome-only hero — saved evidence sitting on a neutral report sheet.
 */
@Composable
fun WelcomeEvidenceStack(
    tagsVisible: Boolean,
    reduceMotion: Boolean,
    modifier: Modifier = Modifier,
    photoHeight: Dp = 196.dp,
    drawableResId: Int = R.drawable.welcome_kitchen_main,
) {
    val cardRadius = 28.dp
    val photoRadius = 23.dp
    val sheetOffset = 10.dp

    Box(modifier = modifier.fillMaxWidth()) {
        Surface(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = sheetOffset + 6.dp, start = sheetOffset + 4.dp, end = 2.dp, bottom = 2.dp)
                .height(photoHeight + 56.dp),
            shape = RoundedCornerShape(cardRadius - 2.dp),
            color = MMColors.PaperSurface.copy(alpha = 0.12f),
            border = androidx.compose.foundation.BorderStroke(0.75.dp, MMColors.CardStroke.copy(alpha = 0.35f)),
        ) {}

        Surface(
            modifier = Modifier
                .fillMaxWidth()
                .shadow(6.dp, RoundedCornerShape(cardRadius), ambientColor = Color.Black.copy(0.22f)),
            shape = RoundedCornerShape(cardRadius),
            color = MMColors.EvidenceCard,
            border = androidx.compose.foundation.BorderStroke(1.dp, MMColors.CardStroke.copy(alpha = 0.9f)),
        ) {
            Column {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 12.dp, vertical = 12.dp)
                        .height(photoHeight),
                ) {
                    Box(
                        modifier = Modifier
                            .fillMaxSize()
                            .clip(RoundedCornerShape(photoRadius))
                            .background(MMColors.FieldFill),
                    ) {
                        Image(
                            painter = painterResource(drawableResId),
                            contentDescription = null,
                            contentScale = ContentScale.Crop,
                            modifier = Modifier.fillMaxSize(),
                        )
                        Box(
                            modifier = Modifier
                                .fillMaxWidth()
                                .height(48.dp)
                                .align(Alignment.BottomCenter)
                                .background(
                                    androidx.compose.ui.graphics.Brush.verticalGradient(
                                        listOf(Color.Transparent, Color.Black.copy(alpha = 0.34f)),
                                    ),
                                ),
                        )
                    }

                    Text(
                        "Kitchen · Apr 14 · 5:42 PM",
                        fontSize = 10.sp,
                        fontWeight = FontWeight.Medium,
                        fontFamily = FontFamily.Monospace,
                        color = Color.White.copy(alpha = 0.92f),
                        maxLines = 1,
                        modifier = Modifier
                            .align(Alignment.TopStart)
                            .padding(start = 12.dp, top = 12.dp)
                            .background(Color.Black.copy(alpha = 0.55f), RoundedCornerShape(6.dp))
                            .border(0.5.dp, Color.White.copy(alpha = 0.12f), RoundedCornerShape(6.dp))
                            .padding(horizontal = 8.dp, vertical = 5.dp),
                    )

                    if (tagsVisible) {
                        Column(
                            modifier = Modifier
                                .align(Alignment.BottomStart)
                                .padding(start = 12.dp, bottom = 12.dp, end = 8.dp),
                        ) {
                            issueCallouts.forEachIndexed { index, label ->
                                val tagScale by animateFloatAsState(
                                    if (tagsVisible) 1f else 0.94f,
                                    MMMotion.welcomeTagEnterSpec(reduceMotion),
                                    label = "evidenceCallout$index",
                                )
                                Text(
                                    label,
                                    fontSize = 9.5.sp,
                                    fontWeight = FontWeight.Medium,
                                    color = Color.White.copy(alpha = 0.9f),
                                    maxLines = 1,
                                    modifier = Modifier
                                        .scale(tagScale)
                                        .padding(top = if (index == 0) 0.dp else 5.dp)
                                        .background(Color.Black.copy(alpha = 0.62f), RoundedCornerShape(5.dp))
                                        .border(0.5.dp, Color.White.copy(alpha = 0.14f), RoundedCornerShape(5.dp))
                                        .padding(horizontal = 7.dp, vertical = 3.dp),
                                )
                            }
                        }
                    }
                }

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(MMColors.FieldFill.copy(alpha = 0.38f))
                        .padding(horizontal = 14.dp, vertical = 11.dp),
                    verticalAlignment = Alignment.Top,
                ) {
                    Icon(
                        Icons.Filled.CheckCircle,
                        contentDescription = null,
                        tint = MMColors.Primary.copy(alpha = 0.85f),
                        modifier = Modifier
                            .size(18.dp)
                            .padding(top = 1.dp),
                    )
                    Spacer(Modifier.width(8.dp))
                    Column(Modifier.weight(1f)) {
                        Text(
                            "Saved proof record",
                            fontSize = 14.sp,
                            fontWeight = FontWeight.SemiBold,
                            color = MMColors.TextPrimary,
                            maxLines = 1,
                        )
                        Text(
                            "12 photos attached · Move-in inspection",
                            fontSize = 11.5.sp,
                            fontWeight = FontWeight.Medium,
                            color = MMColors.TextMuted,
                            modifier = Modifier.padding(top = 2.dp),
                            maxLines = 2,
                        )
                    }
                    Text(
                        "Report-ready",
                        fontSize = 10.sp,
                        fontWeight = FontWeight.Medium,
                        color = MMColors.TextMuted.copy(alpha = 0.85f),
                        modifier = Modifier.padding(top = 2.dp),
                        maxLines = 1,
                    )
                }
            }
        }
    }
}
